lib/rollbar/delay/thread.rb

```diff
@@ -1,9 +1,61 @@
+require 'thread'
+require 'timeout'
+
 module Rollbar
   module Delay
     class Thread
-      def self.call(payload)
-        new.call(payload)
-      end
+      Error        = Class.new(StandardError)
+      TimeoutError = Class.new(Error)
+
+      class << self
+        def call(payload)
+          spawn_threads_reaper
+          threads << new.call(payload)
+        end
+
+        private
+
+        def threads
+          @threads ||= Queue.new
+        end
+
+        def reaper
+          @reaper ||=
+            Thread.start do
+              loop do
+                thread = threads.pop
+
+                if thread == :exit
+                  break
+                else
+                  thread.join
+                end
+              end
+            end
+        end
+
+        def spawn_threads_reaper
+          unless @spawned
+            @spawned = true
+
+            reaper # we spawn the reaper thread before configuring it
```

Comment:
I'd prefer if here we have `@reaper||= build_reaper_thread`. And you use `attr_reader :reaper` in the class definition.